src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -1877,7 +1912,12 @@ def _load_checkpoint_to_model(self):
                 )
 
         # UPDATE TRAINING PARAMS IF THEY EXIST & WE ARE NOT LOADING AN EXTERNAL MODEL's WEIGHTS
-        self.best_metric = self.checkpoint["acc"] if "acc" in self.checkpoint.keys() else -1
+        _best_ckpt_metrics = get_param(self.checkpoint, "_best_ckpt_metrics")
+        if self.load_checkpoint or resume_path:
+            if _best_ckpt_metrics is None or self.metric_to_watch != _best_ckpt_metrics["tracked_metric_name"]:
+                self._eval_before_resume = True
```

Comment:
Is it possible to call `eval` right here and not splitting logic into two parts (detecting whether evaluation is needed and actual evaluation)?

Reply:
Problem is there's alot more that needs to be done before we can call `eval` (instantiating metric objects, loss etc).
I rather not change the order of things in Trainer for this cause.
I can move this check right before calling eval (see why we need it prior to resuming in previous comment) if the flag is bothering.